### DISCUSSION TITLE
If/how to enhance (bluetooth) security?

### DISCUSSION THREAD

**Discussion by 7wells** [fl4p/batmon-ha#146] [Q&A] [answered]
Hi Fabian and all,

Can we please share thoughts if/how to enhance (bluetooth) security? I hope this is the right place to ask.

To my current understanding, many (most?) BMS can be connected to without provision of a PIN. This is definitely the case with my Liontron LiFePO4 battery that has a BMS from JBD. The `batmon` addon can connect to it just by providing the MAC address in the config.

Is it correct that if `concurrent_sampling` is disabled and as long as `batmon` is connected to this (or any other) BMS, no third-party app/ tool can connect to the BMS? What about the `keep_alive` parameter? Does it play a role in this regard - how should it be set up to enhance security? (if it matters)

If security cannot be enhanced programmatically (and I would not be surprised, as it seems to be a failure of the hardware/ BMS concept and has nothing to do with your add-on), then please accept my apologies for asking and close this as you see appropriate.

Thank you very much for your great tool and offering a place to talk about this. 🙂

Cheers
7wells

PS: You can reply also in German. 😀

**Comment by fl4p** [comment, marked as answer]
Hi,
Batmon stays permanently connected with `keep_alive` enabled, no matter the other settings (`concurrent_sampling`).
This will increase privacy, as no one else can connect while batmon is connected.

Of course you can disconnect the dongle (for JK BMS you need to de-solder it).

As the dongle is usually connected to a serial UART (?), you might be able to use a different, secure dongle, thats implements the PSK auth (PIN).

**Reply by 7wells** [reply]
Thank you very much, and please excuse me for my late reply.

I don't know where the dongle is. Is it usually part of the battery ("Akku") or a separate part?

**Reply by fl4p** [reply]
The BLE dongle / adapter is usually inside the battery enclosure, either connected with wires to the BMS mainboard or directly soldered on to it. It should be  easy to spot since it has a small PCB antenna.
